app/code/Magento/InventoryElasticsearch/Plugin/CatalogSearch/Model/Indexer/Fulltext/Action/DataProvider/StockedProductFilterByInventoryStock.php

```diff
@@ -0,0 +1,204 @@
+use Magento\CatalogInventory\Api\StockStatusRepositoryInterface;
+use Magento\CatalogSearch\Model\Indexer\Fulltext\Action\DataProvider;
+use Magento\Elasticsearch\Model\Config;
+use Magento\Framework\App\ResourceConnection;
+use Magento\InventoryCatalogApi\Api\DefaultStockProviderInterface;
+use Magento\InventoryIndexer\Model\StockIndexTableNameResolverInterface;
+use Magento\InventorySalesApi\Model\StockByWebsiteIdResolverInterface;
+
+/**
+ * Filter products by stock status.
+ */
+class StockedProductFilterByInventoryStock
+{
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @var StockConfigurationInterface
+     */
+    private $stockConfiguration;
+
+    /**
+     * @var StockIndexTableNameResolverInterface
+     */
+    private $stockIndexTableNameResolver;
+
+    /**
+     * @var ResourceConnection
+     */
+    private $resourceConnection;
+
+    /**
+     * @var StockByWebsiteIdResolverInterface
+     */
+    private $stockByWebsiteIdResolver;
+
+    /**
+     * @var StockStatusCriteriaInterfaceFactory
+     */
+    private $stockStatusCriteriaFactory;
+
+    /**
+     * @var StockStatusRepositoryInterface
+     */
+    private $stockStatusRepository;
+
+    /**
+     * @var DefaultStockProviderInterface
+     */
+    private $defaultStockProvider;
+
+    /**
+     * @param Config $config
+     * @param StockConfigurationInterface $stockConfiguration
+     * @param StockIndexTableNameResolverInterface $stockIndexTableNameResolver
+     * @param ResourceConnection $resourceConnection
+     * @param StockByWebsiteIdResolverInterface $stockByWebsiteIdResolver
+     * @param StockStatusCriteriaInterfaceFactory $stockStatusCriteriaFactory
+     * @param StockStatusRepositoryInterface $stockStatusRepository
+     * @param DefaultStockProviderInterface $defaultStockProvider
+     */
+    public function __construct(
+        Config $config,
+        StockConfigurationInterface $stockConfiguration,
+        StockIndexTableNameResolverInterface $stockIndexTableNameResolver,
+        ResourceConnection $resourceConnection,
+        StockByWebsiteIdResolverInterface $stockByWebsiteIdResolver,
+        StockStatusCriteriaInterfaceFactory $stockStatusCriteriaFactory,
+        StockStatusRepositoryInterface $stockStatusRepository,
+        DefaultStockProviderInterface $defaultStockProvider
+    ) {
+        $this->config = $config;
+        $this->stockConfiguration = $stockConfiguration;
+        $this->stockIndexTableNameResolver = $stockIndexTableNameResolver;
+        $this->resourceConnection = $resourceConnection;
+        $this->stockByWebsiteIdResolver = $stockByWebsiteIdResolver;
+        $this->stockStatusCriteriaFactory = $stockStatusCriteriaFactory;
+        $this->stockStatusRepository = $stockStatusRepository;
+        $this->defaultStockProvider = $defaultStockProvider;
+    }
+
+    /**
+     * Filter out of stock options for configurable product.
```

Comment:
please remove "of"
and just leave 
`Filter out stock options for configurable product.`